### PR TITLE
fix: N-07 Duplicate Code

### LINKED ISSUE #569: audit: N-07 Duplicate Code
The default contract address is implemented in user_address.cairo and
proposal.cairo .
Consider removing the redundant version.

### PR DESCRIPTION
**Pull request by pscott**
Closes #569

### CODE REVIEW

File: starknet/src/types/proposal.cairo [hunk 1]
```diff
@@ -1,6 +1,9 @@
 use starknet::{ContractAddress, storage_access::StorePacking, Store, contract_address_const};
 use sx::{
-    utils::math::pow, types::{FinalizationStatus, UserAddress, user_address::UserAddressTrait}
+    utils::math::pow,
+    types::{
+        FinalizationStatus, UserAddress, user_address::{UserAddressTrait, ContractAddressDefault}
```
Comment (Orland0x) [inline starknet/src/types/proposal.cairo]: we re-exported `user_address` so shouldnt need the full import path

File: starknet/src/types/proposal.cairo [hunk 2]
```diff
@@ -43,13 +46,6 @@ struct Proposal {
     active_voting_strategies: u256
 }
 
-// Required for the proposal derivation. Ideally, ContractAddress would impl Default in the corelib.
-impl ContractAddressDefault of Default<ContractAddress> {
```
Comment (Orland0x) [inline starknet/src/types/proposal.cairo:47]: might make more sense to put this in a dedicated file like `defaults.cairo` in `/utils` as its kind of random here. We have `into.cairo` which  serves similar role for the `Into` trait (impls of standard lib types that are not (yet) in the standard lib).

File: starknet/src/types/proposal.cairo (hunk 1 above)
Comment (pscott) [reply starknet/src/types/proposal.cairo]: 022db4eec4c3cb91d24b7f99eb679ea1771c2039

File: starknet/src/types/proposal.cairo (hunk 2 above)
Comment (pscott) [reply starknet/src/types/proposal.cairo:47]: true! as I said, ideally, ContractAddress would have a default impl in core lol

7acb81eaa22fab7371b982c8da3358fc70c570d4

Comment (Orland0x) [reply starknet/src/types/proposal.cairo:47]: agreed. `defaults.cairo` is in `types` not `utils` also dont think the impl has been removed from `user_address.cairo`

File: starknet/src/types/default.cairo [hunk 3]
```diff
@@ -0,0 +1,8 @@
+use starknet::{ContractAddress, contract_address_const};
+
+// Required for the proposal derivation. Ideally, ContractAddress would impl Default in the corelib.
```
Comment (Orland0x) [inline starknet/src/types/default.cairo]: comment no longer really applicable

File: starknet/src/types/proposal.cairo (hunk 2 above)
Comment (pscott) [reply starknet/src/types/proposal.cairo:47]: 1872fed814ed4136e3d6e63eb282a2a3ae0a3ea6

File: starknet/src/types/default.cairo (hunk 3 above)
Comment (pscott) [reply starknet/src/types/default.cairo]: 1872fed814ed4136e3d6e63eb282a2a3ae0a3ea6

**Review by Orland0x — APPROVED** [review]
uTACK

**Pull request merged**